src/osd/PG.cc

```diff
@@ -4645,7 +4693,13 @@ void PG::chunky_scrub(ThreadPool::TPHandle &handle)
           // will be requeued by sub_op_scrub_map
           dout(10) << "wait for replicas to build scrub map" << dendl;
           done = true;
-        } else {
+        } else if (scrubber.preempted) {
+	  scrubber.preempt_left--;
+	  scrubber.preempt_divisor *= 2;
+	  dout(10) << __func__ << " preempted, " << scrubber.preempt_left
+		   << " left, restarting chunk" << dendl;
+	  scrubber.state = PG::Scrubber::NEW_CHUNK;
```

Comment:
scrubber.state became NEW_CHUNK by preempted then restart a new chunk scrub, but the  MSG_OSD_REP_SCRUBMAP msg of the previous scrub may comes,  scrubber.waiting_on_whom will erase error in do_replica_scrub_map . should it discard old MSG_OSD_REP_SCRUBMAP.?